src/routes/endpoints.py

```diff
@@ -0,0 +1,44 @@
+from fastapi import APIRouter, UploadFile, File, HTTPException, status
+from src.services.flow_controller_service import FlowService
+from src.exceptions.custom_exceptions import PDFProcessingError
+from src.schemas.files import PDFStatus
+import logging
+import io
+
+router = APIRouter()
+
+logging.basicConfig(level=logging.DEBUG)  
+
+@router.post('/files/flow/{document_id}', status_code=200)
+async def analyze_pdf(document_id: int, pdf_file: UploadFile = File(...)):
+    try:
+        contents = await pdf_file.read() 
+        flow_service = FlowService(contents)
```

Comment:
you shouldn't instantiate classes here, add a dependency.py file and add you instantiations there.

use factory pattern to create classes:

class PdfServiceFactory:
  def create_pdf_service():
     return PDFService()

pdf_service_factory = PdfServiceFactory()


then use the pdf_service_factory in the routers